lib/src/controls/surfaces/commandbar.dart

```diff
@@ -0,0 +1,552 @@
+///
+/// A command bar is composed of a series of [CommandBarItem]s, which each could
+/// be a [CommandBarButton] or a custom [CommandBarItem].
+///
+/// If there is not enough horizontal space to display all items, the overflow
+/// behavior is determined by [overflowBehavior].
+///
+/// ![CommandBar example](https://docs.microsoft.com/en-us/windows/apps/design/controls/images/controls-appbar-icons.png)
+///
+/// See also:
+///
+///   * <https://docs.microsoft.com/en-us/windows/apps/design/controls/command-bar>
+class CommandBar extends StatefulWidget {
+  /// The [CommandBarItem]s that should appear on the primary area.
+  final List<CommandBarItem> primaryItems;
+
+  /// If non-empty, a "overflow item" will appear on the primary area
+  /// (as built by [overflowItemBuilder], or it will be a "more" button
+  /// if [overflowItemBuilder] is null), and when activated, will show a
+  /// flyout containing this list of secondary items.
+  final List<CommandBarItem> secondaryItems;
+
+  /// Allows customization of the "overflow item" that will appear on the
+  /// primary area of the command bar if there are any items in the
+  /// [secondaryItems] (including any items that are dynamically considered
+  /// to be there if [overflowBehavior] is
+  /// [CommandBarOverflowBehavior.dynamicOverflow].)
+  final CommandBarActionItemBuilder? overflowItemBuilder;
+
+  /// Determines what should happen when the items are too wide for the
+  /// primary command bar area. See [CommandBarOverflowBehavior].
+  final CommandBarOverflowBehavior overflowBehavior;
+
+  /// If the width of this widget is less then the indicated amount,
+  /// items in the primary area will be rendered using
+  /// [CommandBarItemDisplayMode.inPrimaryCompact]. If this is null
+  /// or the width of this widget is wider, then the items will be rendered
+  /// using [CommandBarItemDisplayMode.inPrimary].
+  final double? compactBreakpointWidth;
+
+  /// The alignment of the items within the command bar across the main axis
+  final MainAxisAlignment mainAxisAlignment;
+
+  /// The alignment of the items within the command bar across the cross axis
+  final CrossAxisAlignment crossAxisAlignment;
+
+  /// The alignment of the overflow item (if displayed) between the end of
+  /// the visible primary items and the end of the boundaries of this widget.
+  /// Only relevant if [overflowBehavior] is
+  /// [CommandBarOverflowBehavior.dynamicOverflow].
+  final MainAxisAlignment overflowItemAlignment;
+
+  /// The width of the flyout menu used to display secondary commands.
+  final double flyoutWidth;
+
+  final bool _isExpanded;
+
+  const CommandBar({
+    Key? key,
+    required this.primaryItems,
+    this.secondaryItems = const [],
+    this.overflowItemBuilder,
+    this.overflowBehavior = CommandBarOverflowBehavior.dynamicOverflow,
+    this.compactBreakpointWidth,
+    this.mainAxisAlignment = MainAxisAlignment.start,
+    this.crossAxisAlignment = CrossAxisAlignment.center,
```

Comment:
on #181, Right to Left Support was added. The current `Directionality` should be taken into account

Reply:
Tight-to-left Directionality is already considered in the widgets used inside the `CommandBar` (whether it's `Row` or `DynamicOverflow`). I just needed to change the text "right-aligned" to "end-aligned" in the example so it's not confusing when right-to-left mode is enabled:
![image](https://user-images.githubusercontent.com/88254524/159275955-05eee4da-61ce-495e-82ee-a779f5b3f220.png)

For the secondary flyout menu content, I did notice that `ListTile` needed its padding fixed in RTL mode, so I fixed that.